src/components/Tutorial/TutorialSwap/index.tsx

```diff
@@ -0,0 +1,508 @@
+import { SUPPORTED_WALLETS } from 'constants/index'
+import { useActiveWeb3React } from 'hooks'
+import useMixpanel, { MIXPANEL_TYPE } from 'hooks/useMixpanel'
+import useTheme from 'hooks/useTheme'
+import { useTutorialSwapGuide } from 'state/tutorial/hooks'
+import { useIsDarkMode } from 'state/user/hooks'
+
+import CustomMask from './CustomMask'
+import CustomPopup from './CustomPopup'
+import TutorialMobile from './TutorialMobile'
+import { LIST_TITLE, StepCustom, TOTAL_STEP, TutorialIds } from './constant'
+
+const isMobile = window.innerWidth < 1200 // best resolution for this tutorial
+
+const Heading = styled.h5`
+  color: ${({ theme }) => theme.text};
+  user-select: none;
+  margin: 5px 0px 10px 0px;
+  display: flex;
+  align-items: center;
+  font-size: 16px;
+`
+
+const LayoutWrapper = styled.div`
+  color: ${({ theme }) => theme.subText};
+  text-align: left;
+  font-size: 14px;
+`
+const Title = ({ stepNumber }: { stepNumber: number }) => {
+  const theme = useTheme()
+  return (
+    <Heading style={{ display: 'flex', alignItems: 'flex-end' }}>
+      <span>
+        <Trans>Step</Trans>: {stepNumber}/
+      </span>
+      <span style={{ color: theme.subText, fontSize: '0.85em' }}>{TOTAL_STEP}</span>
+    </Heading>
+  )
+}
+
+const Layout = ({ children, title }: { title?: string; children: React.ReactNode }) => {
+  return (
+    <LayoutWrapper>
+      {!isMobile && title && <Heading>{title}</Heading>}
+      {children}
+    </LayoutWrapper>
+  )
+}
+
+const ArrowWrapper = styled.div`
+  width: 32px;
+  height: 32px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  color: ${({ theme }) => theme.text};
+  svg {
+    transition: all 150ms ease-in-out;
+  }
+  &[data-expanded='false'] {
+    svg {
+      transform: rotate(180deg);
+    }
+  }
+`
+
+const NetworkItemWrapper = styled.div`
+  background: ${({ theme }) => theme.buttonBlack};
+  border-radius: 42px;
+  display: flex;
+  padding: 10px 15px;
+  gap: 10px;
+  cursor: pointer;
+`
+
+const NetworkWrapper = styled.div`
+  background: ${({ theme }) => theme.background};
+  border-radius: 20px;
+  padding: 15px;
+  gap: 10px;
+  display: flex;
+  flex-direction: column;
+`
+
+const ImageMobile = ({ imageName, marginTop = false }: { imageName: string; marginTop?: boolean }) =>
+  isMobile ? (
+    <Flex justifyContent={'center'}>
+      <img
+        style={{ marginTop: marginTop ? 20 : 0, width: '100%', maxWidth: 800 }}
+        src={require(`../../../assets/images/tutorial_swap/${imageName}`).default}
+        alt={imageName}
+      />
+    </Flex>
+  ) : null
+
+const Desc = styled.p`
+  line-height: 20px;
+`
+
+const HighlightText = styled.span`
+  color: ${({ theme }) => theme.text};
+`
+function Welcome() {
+  return (
+    <Layout>
+      <img src={WelcomeImage} alt="welcome to kyberswap" style={{ maxWidth: '100%', marginTop: 10 }} />
+      <Desc>
+        KyberSwap is a decentralized exchange (DEX) aggregator. We provide our traders with the{' '}
```

Comment:
remove redundant `{' '}`

Reply:
```
KyberSwap is a decentralized exchange (DEX) aggregator. We provide our traders with the{' '}
<HighlightText>best token prices</HighlightText>
```
it's not redundant. it will render a space.